### PR TITLE
fix: wait for initializationPromise

### PR DESCRIPTION
**Pull request by SophieDePaula**
[Trello card](https://trello.com/c/Ww0h1hQV/1884-storybook-race-condition-leading-to-failed-tests)
[I also see that Storybook awaits on `initializationPromise` before calling `setSelection`](https://github.com/storybookjs/storybook/blob/992efdfec892e86fb85ef47e9c5ade53180f40ca/code/lib/preview-api/src/modules/preview-web/PreviewWithSelection.tsx#L223)

### CODE REVIEW

File: js/packages/eyes-storybook/src/browser/getClientAPI.js [hunk 1]
```diff
@@ -138,13 +139,13 @@ function getClientAPI() {
 }
 
 function getFrameWindow() {
-  if (/iframe.html/.test(window.location.href)) {
+  if (/iframe/.test(window.location.href)) {
```
Comment (amitzur) [inline js/packages/eyes-storybook/src/browser/getClientAPI.js:142]: this is unrelated to the current PR, why is it here?

File: js/packages/eyes-storybook/src/browser/getClientAPI.js [hunk 2]
```diff
@@ -138,13 +139,13 @@ function getClientAPI() {
...
     return window;
   }
 
   const innerFrameWindow = Array.prototype.find.call(window.frames, frame => {
     try {
-      return /\/iframe.html/.test(frame.location.href);
+      return /\/iframe/.test(frame.location.href);
```
Comment (amitzur) [inline js/packages/eyes-storybook/src/browser/getClientAPI.js:148]: same as last comment

File: js/packages/eyes-storybook/test/fixtures/storybook-versions/latest/package.json [hunk 3]
```diff
@@ -4,10 +4,10 @@
   "private": true,
   "scripts": {
     "build": "react-scripts build",
-    "build-storybook": "build-storybook",
+    "build-storybook": "sb build",
     "eject": "react-scripts eject",
     "start": "react-scripts start",
-    "storybook": "APPLITOOLS_FRAMEWORK_VERSION=latest start-storybook -p 9010 --no-manager-cache",
+    "storybook": "STORYBOOK_VERSION=latest sb dev -p 9010",
```
Comment (amitzur) [inline js/packages/eyes-storybook/test/fixtures/storybook-versions/latest/package.json:10]: I don't understand. Did latest not run Storybook 7 before? Also, `STORYBOOK_VERSION` is not read anywhere. And `APPLITOOLS_FRAMEWORK_VERSION` is the way to set storybook version now. So what is this change?

**Review by amitzur — CHANGES_REQUESTED** [review]
Did you manage to reproduce this?
There is another direction of using the channel to actually emit the `SET_CURRENT_STORY` event to render the story:
https://github.com/storybookjs/storybook/blob/97a2b8940112dd62c2775feb1fe9bafb93bf0bd2/code/lib/preview-api/src/modules/preview-web/PreviewWithSelection.tsx#L101
But that's a bigger change that requires more investigation. The solution in this PR is good enough IMO

File: js/packages/eyes-storybook/src/browser/getClientAPI.js (hunk 1 above)
Comment (SophieDePaula) [reply js/packages/eyes-storybook/src/browser/getClientAPI.js]: Yes, this is a different issue. I removed it and will address it in a different PR.

File: js/packages/eyes-storybook/src/browser/getClientAPI.js (hunk 2 above)
Comment (SophieDePaula) [reply js/packages/eyes-storybook/src/browser/getClientAPI.js]: I responded on the last comment

File: js/packages/eyes-storybook/test/fixtures/storybook-versions/latest/package.json (hunk 3 above)
Comment (SophieDePaula) [reply js/packages/eyes-storybook/test/fixtures/storybook-versions/latest/package.json]: @amitzur I changed it to `APPLITOOLS_FRAMEWORK_VERSION`, and not it didn't use 7 I had to manually update some dependencies to make this run on 7

**Comment by SophieDePaula** [comment]
> Did you manage to reproduce this? There is another direction of using the channel to actually emit the `SET_CURRENT_STORY` event to render the story: https://github.com/storybookjs/storybook/blob/97a2b8940112dd62c2775feb1fe9bafb93bf0bd2/code/lib/preview-api/src/modules/preview-web/PreviewWithSelection.tsx#L101 But that's a bigger change that requires more investigation. The solution in this PR is good enough IMO

I was not able to reproduce this, can I merge this?

**Comment by SophieDePaula** [comment]
@kyrylodolynskyi I added you as a reviewer as I see that Amit is on PTO

**Review by amitzur — APPROVED** [review]

**Pull request merged**